### PR TITLE
Add some info about `documentation/samplefiles*` and `documentation/api-scenario`

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
As in subject. At first I planned to delete the `samplefiles*` and `api-scenario` directories, but conversation of this PR led me to instead update the information about them.

Related work:
- https://github.com/Azure/azure-sdk-tools/issues/7628
- https://github.com/Azure/azure-rest-api-specs/pull/28464
- [Pull Request 953](https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-docs-eng.ms/pullrequest/953): Forward specs dir layout article to specs repo article; add links to APEX breaking changes; update AutoRest links
- [Pull Request 963](https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-docs-eng.ms/pullrequest/963): Update link to sample AutoRest README.md + clarify meaning of branch protection rules table

### CODE REVIEW

**Comment by maririos** [comment]
As a sanity check, adding @raych1 for the SDK automation references

File: documentation/api-scenario/how-to/GenerateABasicApiScenario.md [hunk 1]
```diff
@@ -1,36 +0,0 @@
-# Generate a basic API Scenario file
```
Comment (maririos) [inline documentation/api-scenario/how-to/GenerateABasicApiScenario.md:1]: wow cool that we can remove this. I was not aware of these files

Comment (maririos) [reply documentation/api-scenario/how-to/GenerateABasicApiScenario.md:1]: wait... actually, I think we might need this... checking the release planner

Comment (maririos) [reply documentation/api-scenario/how-to/GenerateABasicApiScenario.md:1]: yeah.. The release planner does:

image

The first link points to: https://github.com/Azure/azure-rest-api-specs/tree/main/documentation/api-scenario

@raych1 @josefree @ArthurMa1978  is this something we should keep asking service teams to use/do?

Comment (konrad-jamrozik) [reply documentation/api-scenario/how-to/GenerateABasicApiScenario.md:1]: Also @rkmanda @AkhilaIlla as you folks work on [Swagger live validation](https://devdiv.visualstudio.com/DevDiv/_git/openapi-swaggerkpi) and I wonder if these two tools are connected.

@weshaggard FYI

File: documentation/api-scenario/how-to/QuickStart.md [hunk 2]
```diff
@@ -1,150 +0,0 @@
...
-
- This software is released under the MIT License.
- https://opensource.org/licenses/MIT
--->
-
-# Quick start with API Scenario test
-
-## Install
-
-`oav` is an open-source powerful tool for swagger validation, API Scenario testing, and examples generation. GitHub: https://github.com/Azure/oav.
```
Comment (konrad-jamrozik) [inline documentation/api-scenario/how-to/QuickStart.md:12]: @scbedd given that you own `oav`: do you recall if it is still being used to power `api-scenario` ? If so, can you have a look on these docs to assess if they are current?

Comment (weshaggard) [reply documentation/api-scenario/how-to/QuickStart.md:12]: Yes I know OAV supports example generation. It is also used to produce coverage reports for specs.

Comment (konrad-jamrozik) [reply documentation/api-scenario/how-to/QuickStart.md:12]: @weshaggard so we still use `api-scenario` functionality, yes?

Comment (konrad-jamrozik) [reply documentation/api-scenario/how-to/QuickStart.md]: In any case, I reverted deletions in `documentation/api-scenario/`.

**Review by weshaggard — APPROVED** [review]

File: documentation/api-scenario/how-to/GenerateABasicApiScenario.md (hunk 1 above)
Comment (raych1) [reply documentation/api-scenario/how-to/GenerateABasicApiScenario.md]: @maririos we ever discussed on leveraging API scenario to create test in TypeSpec scenario but haven't concluded yet. //CC: @lirenhe

File: documentation/samplefiles/Microsoft.YourServiceName/stable/YYYY-MM-DD/scenarios/quickStart.yaml [hunk 3]
```diff
@@ -1,34 +0,0 @@
-# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/documentation/api-scenario/references/v1.2/schema.json
```
Comment (raych1) [inline documentation/samplefiles/Microsoft.YourServiceName/stable/YYYY-MM-DD/scenarios/quickStart.yaml:1]: This doc is used as the quick start for API scenarios. If you keep the API scenario docs, this one should be reverted too.

**Review by raych1 — COMMENTED** [review]
@konrad-jamrozik , can you share me where the sample readme goes?

**Comment by konrad-jamrozik** [comment]
> @konrad-jamrozik , can you share me where the sample readme goes?

@raychen1 I am unsure what you mean. I didn't delete the sample readme, which is here:

https://github.com/Azure/azure-rest-api-specs/blob/users/kojamroz/del_obs_samples/documentation/samplefiles/samplereadme.md

**Comment by raych1** [comment]
> > @konrad-jamrozik , can you share me where the sample readme goes?
> 
> @raychen1 I am unsure what you mean. I didn't delete the sample readme, which is here:
> 
> * https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/samplefiles/samplereadme.md
> * https://github.com/Azure/azure-rest-api-specs/blob/users/kojamroz/del_obs_samples/documentation/samplefiles/samplereadme.md

@konrad-jamrozik , thanks. Can you keep the files under `documentation/samplefiles-dp/`? The readme files for both multiple clients and single client are still valid.

File: documentation/api-scenario/how-to/GenerateABasicApiScenario.md (hunk 1 above)
Comment (konrad-jamrozik) [reply documentation/api-scenario/how-to/GenerateABasicApiScenario.md]: Keeping `api-scenario`. Just updated README.md a little bit.

**Pull request merged**